### ISSUE TITLE
More clangformat.exe information needed

### ISSUE THREAD

**Issue by shanholtzerk** [MicrosoftDocs/visualstudio-docs#7989]
Please specify where clangformat.exe resides and whether it supports the command to create a clang-format file, such as, clang-format -style=llvm -dump-config > .clang-format.
Also, it's not clear whether the VS clangformat.exe supports all clang-format functionality. 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 9b9340cc-8322-6f0f-2e30-1c5e18667a87
* Version Independent ID: dc327ebb-0d9c-f9f1-3465-5acc6cee1bc6
* Content: [Options, Text Editor, C/C++, Formatting - Visual Studio (Windows)](https://docs.microsoft.com/en-us/visualstudio/ide/reference/options-text-editor-c-cpp-formatting?view=vs-2019)
* Content Source: [docs/ide/reference/options-text-editor-c-cpp-formatting.md](https://github.com/MicrosoftDocs/visualstudio-docs/blob/main/docs/ide/reference/options-text-editor-c-cpp-formatting.md)
* Product: **visual-studio-windows**
* GitHub Login: @TerryGLee
* Microsoft Alias: **tglee**

**Comment by WilliamAntonRohm** [comment]
@shanholtzerk -- thank you for your feedback.  Please consider these resources:
- [Visual Studio on Microsoft Q&A](https://docs.microsoft.com/answers/products/vs) 
- [Visual Studio MSDN Forum](https://social.msdn.microsoft.com/Forums/vstudio) 
- [Stack Overflow](https://stackoverflow.com/search?q=%22visual+studio%22)  
- [Reddit](https://www.reddit.com/r/VisualStudio)

**Comment by corob-msft** [comment]
@shanholtzerk 
It's possible that not every single formatting possibility is represented in the IDE, but it's not because Visual Studio is using a special version of clang-format. You should be able to use any functionality with the tool itself.

When installed by Visual Studio, the clang tools are in *`%VCINSTALLDIR%Tools\Llvm\bin`* and  *`%VCINSTALLDIR%Tools\Llvm\x64\bin`*. These are the regular llvm/clang tools, updated approximately with every Visual Studio long-term servicing release (so they might be behind the official mainstream release by as much as six months, depending on when you check). This is what I see when I check the version on clang-format.exe:

```
C:\Program Files\Microsoft Visual Studio\2022\Enterprise>"C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\Llvm\x64\bin\clang-format.exe" --version
clang-format version 13.0.1
```

So, that looks like it's one version back from the latest.

**Comment by shanholtzerk** [comment]
@WilliamAntonRohm I had already searched those sources except for Reddit. @corob-msft , thank you for that information. I was able to dump the current config.
The referenced VS page, https://docs.microsoft.com/en-us/visualstudio/ide/reference/options-text-editor-c-cpp-formatting?view=vs-2022, has this sentence:
"By default, Visual Studio runs clangformat.exe in the background applies formatting as you type. "
It should read:
"By default, Visual Studio runs clang-format.exe in the background, applying formatting as you type. "
This corrects the referenced exe filename and the sentence grammar.
